shm_wrapper/shm_wrapper.c

```diff
@@ -535,10 +536,16 @@ void device_disconnect (int dev_ix)
 	//update the catalog
 	dev_shm_ptr->catalog &= (~(1 << dev_ix));
 
-	//reset param map values to 0
+	//reset bitmap values to 0
 	dev_shm_ptr->cmd_map[0] &= (~(1 << dev_ix));   //reset the changed bit flag in cmd_map[0]
 	dev_shm_ptr->cmd_map[dev_ix + 1] = 0;          //turn off all changed bits for the device
 
+	dev_shm_ptr->exec_sub_map[0] |= 1 << dev_ix;
+	dev_shm_ptr->exec_sub_map[dev_ix + 1] = -1;
+
+	dev_shm_ptr->net_sub_map[0] |= 1 << dev_ix;
+	dev_shm_ptr->net_sub_map[dev_ix + 1] = -1;
+
```

Comment:
So correct me if I'm wrong but on device disconnect we reset the subscriptions to all parameters for that device index?

Reply:
If a device with shared memory index `i` disconnects and we reset the subscriptions to `-1` , then the next device assigned to the same shm index will start with a subscription to all params right?

I guess that would be safer in the sense that we have all possible incoming data immediately and we tell dev handler what data we don't need. But I also think setting them to all 0 would save data encoding time... though not by much because net handler and executor will probably send updated subscription requests within a second, right?

Reply:
So net_handler has to always reset to -1 since otherwise Dawn would never know what params exist/can be selected. Unless we are assuming that they also have a copy of the devices within `runtime_util.c` . I'm note sure if they should since their API should be as abstract as possible to allow any device and any param.

Reply:
Ok that makes sense. We should make a note that once Dawn implements the subscriptions, we should change this back to setting to 0 by default, since then Dawn will have the ability to change this bitmap.

Reply:
No regardless it needs to be -1. How would dawn know what parameters the device has and to subscribe to?

Reply:
@AVSurfer123  Since all the values in the protobufs are optional, by default we can be sending DevData messages to Dawn that don't have any of the `ival` `fval` or `bval` fields set. We only send Dawn the list of connected devices (with name, uid, and type), and then each connected device will have a list of all of their parameters (which only contain names). That's enough information for Dawn to display the dashboard of all devices with all parameter names collapsed.

Reply:
oh right now if you don't subscribe you don't get the param name, because the name is part of the Param protobuf struct